src/full-screen/components/runtime.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useCallback, useEffect } from "react";
+import _screenfull from "screenfull";
+import queryString from "query-string";
+import { FullScreenButton } from "./full-screen-button";
+import { useForceUpdate } from "../../shared/hooks/use-force-update";
+
+import css from "./runtime.scss";
+
+const screenfull = _screenfull.isEnabled ? _screenfull : undefined;
+
+// eslint-disable-next-line no-empty-pattern
+export const Runtime: React.FC = ({}) => {
+  const forceUpdate = useForceUpdate();
+  const toggleFullScreen = useCallback(() => {
+    screenfull?.toggle();
+  },[]);
+
+  useEffect(() => {
+    const onChange = () => forceUpdate();
+    screenfull?.on("change", onChange);
+    window.addEventListener('resize', onChange);
+    return () => {
+      screenfull?.off("change", onChange);
+      window.removeEventListener('resize', onChange);
+    };
+  }, [forceUpdate]);
+
+  const isFullScreen = screenfull?.isFullscreen;
```

Comment:
I'm not sure exactly where the best place to do this would be, but I think it would be worth adding a comment about where this code came from, e.g. `This code is patterned after the jQuery-based implementation in the [Cloud File Manager](... link to CFM repo ...).`